### PR TITLE
Feat saml sso

### PR DESCRIPTION
**Pull request by JimMoen**
Fixes <issue-or-jira-number>

### <samp>🤖 Generated by Copilot at e09b2cd</samp>

This pull request adds support for SAML authentication in the `emqx_dashboard_sso` app, by introducing a new module `emqx_dashboard_sso_saml` and updating the related APIs, schemas, and configurations. It also updates the code owners and the license of the `emqx_dashboard_rbac` app, and fixes a dependency issue in the `rebar.config` files.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by HJianBo — APPROVED** [review]

**Review by zhongwencool — APPROVED** [review]

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_saml.erl [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+%%------------------------------------------------------------------------------
+%% APIs
+%%------------------------------------------------------------------------------
+
+create(#{sp_sign_request := true} = Config) ->
+    try
+        do_create(ensure_cert_and_key(Config))
+    catch
+        Kind:Error ->
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_saml.erl:100]: :Stacktrace

or
throw:Reason for unknown reasons.

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_saml.erl [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            name = "EMQX",
+            email = "contact@emqx.io"
+        }
+    }),
+    try
+        IdpMeta = esaml_util:load_metadata(binary_to_list(IDPMetadataURL)),
+        State = Config,
+        {ok, State#{idp_meta => IdpMeta, sp => SP}}
+    catch
+        Kind:Error ->
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_saml.erl:142]: catch known errors, and log known errors only.
for unknown errors, log stacktrace too.

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_saml.erl [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+            ],
+            {error, iolist_to_binary(Reason)}
+    end.
+
+do_validate_assertion(SP, DuplicateFun, Body) ->
+    PostVals = cow_qs:parse_qs(Body),
+    SAMLEncoding = proplists:get_value(<<"SAMLEncoding">>, PostVals),
+    SAMLResponse = proplists:get_value(<<"SAMLResponse">>, PostVals),
+    RelayState = proplists:get_value(<<"RelayState">>, PostVals),
+    case (catch esaml_binding:decode_response(SAMLEncoding, SAMLResponse)) of
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_saml.erl:193]: use try catch

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_saml_api.erl [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+sp_saml_callback(post, Req) ->
+    case emqx_dashboard_sso_manager:lookup_state(saml) of
+        undefined ->
+            {404, #{code => ?BACKEND_NOT_FOUND, message => <<"Backend not found">>}};
+        State ->
+            case (provider(saml)):callback(Req, State) of
+                {ok, Role, Token} ->
+                    {200, emqx_dashboard_sso_api:login_reply(Role, Token)};
+                {error, Reason} ->
+                    ?SLOG(info, #{
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_saml_api.erl:102]: warning ?

**Pull request merged**